container-registry-configuration/README.md

```diff
@@ -0,0 +1,189 @@
+
+#### 1.b FairwindsOps/saffire
+
+There is an open-source project to patch pods in `imagePullErrors` error state: https://github.com/FairwindsOps/saffire
+
+You are able to define alternative source for an image with a Custom Resource.
+
+```
+apiVersion: saffire.fairwinds.com/v1alpha1
+kind: AlternateImageSource
+metadata:
+  name: alternateimagesource-sample
+spec:
+  imageSourceReplacements:
+    - equivalentRepositories:
+        - quay.io/gianswarm/our-image
+        - azurecr.io/giantswarm/our-image
+        - docker.io/giantswarm/our-image
+```
+
+This mechanism doesn't work for static pods and it doesn't prevent the issue in advance. Also, it has no built-in mechanism for authentication. We can extend it or combine with the image pull secret solution but it sounds too complicated and dirty.
+
+
+#### 1.c Containerd Configuration
+
+As we do in vintage clusters, we can configure containerd configuration as below. This seems the best approach.
+
+```
+[plugins."io.containerd.grpc.v1.cri".registry]
+
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors."docker.io"]
+endpoint = ["giantswarm.azurecr.io"]
+
+[plugins."io.containerd.grpc.v1.cri".registry.configs]
+[plugins."io.containerd.grpc.v1.cri".registry.configs."registry-1.docker.io".auth]
+username = "giantswarm-user-account"
+password = "my_token_from_docker_hub"
+```
+
+#### Decision
+
+`1.c` is selected.
+We will use containerd configuration. 
+
+### 2. Authentication
+
+#### 2.a Using unauthenticated accounts
+
+Using only public images without authentication is an option to avoid complexity of securing credentials. However, registry providers have pull/rate limits per account. We will be limited by total of [free account limit](https://docs.docker.com/docker-hub/download-rate-limit/) of Docker Hub (100 pulls per 6 hours per IP address) and [tier limit](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-skus) of Azure Container Registry (3000 ReadOps per minute for standard). Anyone can pull images from our registry and spend our limits, which makes us vulnerable. 
+
+#### 2.b Using authenticated accounts
+
+We can pass credentials to `containerd` configuration to be not limited by registry provider limits.
+
+#### Decision
+
+`2.b` is selected.
+We will use authenticated accounts.
+
+
+### 3. Registries
+
+The infranet page (See `Registry Mirrors`) states this as a desired configuration:
+  - `docker.io` as primary public registry since it has a privilege in docker daemon.
+  - Our own Azure Container Registry as secondary registry because of the security concerns
+
+Since we use `containerd`, docker has not a privelege anymore but there is no need to change this order at the moment. We will follow this configuration for CAPI clusters too.
```

Comment:
```suggestion
Since we use `containerd`, docker has not a privilege anymore but there is no need to change this order at the moment. We will follow this configuration for CAPI clusters too.
```

okay cool, ignore above